docs/modules/ROOT/pages/explanations/decisions/local-objectstorage.adoc

```diff
@@ -0,0 +1,100 @@
+
+Most platforms on which we run APPUiO Managed already provides some kind of S3 compatible storage and whenever possible AppCat directly interfaces with it to provide self-service object buckets.
+
+However, in some cases the underlying platform doesn't provide any object storage, and we don't have the option to use an off-site solution of another provider.
+In these cases we still need a self-contained storage solution that can provide S3 compatible storage and self-service object buckets on the platform.
+
+=== Goals
+
+* Provide S3 compatible storage
+* Few dependencies on the underlying platform
+* Self-service provisioning of object buckets
+
+=== Non-Goals
+
+* Provide any other kind of storage
+* Self-service provisioning of the storage solution itself
+
+== Proposals
+
+=== MinIO
+
+https://min.io/[MinIO] is an open-source, S3 compatible object store.
+It can easily be hosted in Kubernetes and can run with limited resources.
+It only needs access to some kind of persistent volumes.
+
+We already have component to deploy MinIO, but it uses legacy chart that has been archived since 2021.
+So we'd have to update the component to use the new Helm chart.
+
+We did not find any provider/operator that allows bucket and user management.
+There is the https://github.com/minio/operator[official Operator] but that operator focuses on deploying and running multiple instances of MinIO itself.
+There also was https://github.com/kubernetes-retired/cosi-driver-minio[the COSI driver MinIO] that was a sample driver for the https://container-object-storage-interface.github.io/[Container Object Storage Interface (COSI) API].
+However, that driver is no longer supported and was never intended to be production ready.
+We suspect MinIO will eventually provide a COSI driver, but as of writing this, there is no operator that let's us manage buckets and users.
+
+We see no advantage in using the MinIO Operator to deploy MinIO itself.
+So we propose to install MinIO using the newer helm chart and component MinIO.
+To create buckets in this MinIO instance, we write a `provider-minio`, which similarly to `provider-cloudscale` and `provider-exoscale` can create buckets and users.
+There is a https://github.com/minio/minio-go[Go SDK] to interact with MinIO, but we might need to use the https://github.com/minio/madmin-go[MinIO Admin Go Client SDK] to create users.
+
+
+==== Concerns
+There are some concerns around the AGPL v3 license of MinIO.
+Some companies such as https://opensource.google/documentation/reference/using/agpl-policy[Google] categorically avoid any software licensed under AGPL as it is fairly restricitve and may force you to release your own code under the AGPL license.
+However, as we do not in any way modify MinIO itself, so we belief using it should be fine, and we don't need to release AppCat under AGPL.
+But we might need to release our provider under the AGPL v3 license, as the MinIO Admin Go Client SDK is also released under AGPL.
```

Comment:
Is the mere interaction with Minio throught the API really enough to force us to use AGPL? This might need to be verified first.

Reply:
No, but the admin SDK itself is released under AGPL and as we compile that into the binary, we'll probably need to release the provider itself under AGPL

Reply:
I want to have this clearly proven, not just guessed. Please check with LEXR about that, you can contact me for details in the chat.

Reply:
As far as I know, we're now sure that if we use the SDK we'll need to release the provider under AGPLv3 to be on the safe side. Otherwise, we don't see any issue with using MinIO. So think this is resolved.